plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -679,6 +679,138 @@ SELECT
+https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-query-store-runtime-stats-transact-sql?view=sql-server-ver15
+*/
+SELECT
+	'sqlserver_azuredb_querystore_runtime_stats' AS [measurement],
+	REPLACE(@@SERVERNAME, '\', ':') AS [sql_instance],
+	DB_NAME() AS [database_name],
+	CONVERT(nvarchar(35), MAX(rsi.start_time), 126) AS interval_start_time,
+	CONVERT(nvarchar(35), MAX(rsi.end_time), 126) AS interval_end_time,
+	CAST(MAX(q.query_id) AS nvarchar(20)) AS query_id,
+	CONVERT(nvarchar(20), MAX(q.query_hash), 1) as query_hash,
+	CAST(rs.plan_id AS nvarchar(20)) AS plan_id,
+	CONVERT(nvarchar(20), MAX(p.query_plan_hash), 1) as query_plan_hash,
+	MAX(rs.max_logical_io_reads) as max_logical_io_reads,
+	SUM(rs.avg_logical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_logical_io_reads,
+	MAX(rs.max_physical_io_reads) as max_physical_io_reads,
+	SUM(rs.avg_physical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_physical_io_reads,
+	MAX(rs.max_logical_io_writes) as max_logical_io_writes,
+	SUM(rs.avg_logical_io_writes * rs.count_executions) / SUM(rs.count_executions) as avg_logical_io_writes,
+	rs.execution_type,
+	MAX(rs.execution_type_desc) AS execution_type_desc,
+	SUM(rs.count_executions) as count_executions,
+	MAX(rs.max_cpu_time) as max_cpu_time,
+	SUM(rs.avg_cpu_time * rs.count_executions) / SUM(rs.count_executions) as avg_cpu_time,
+	MAX(rs.max_dop) as max_dop,
+	SUM(rs.avg_dop * rs.count_executions) / SUM(rs.count_executions) as avg_dop,
+	MAX(rs.max_rowcount) as max_rowcount,
+	SUM(rs.avg_rowcount * rs.count_executions) / SUM(rs.count_executions) as avg_rowcount,
+	MAX(rs.max_query_max_used_memory) AS max_query_max_used_memory,
+	SUM(rs.avg_query_max_used_memory * rs.count_executions) / SUM(rs.count_executions) AS avg_query_max_used_memory,
+	MAX(rs.max_duration) as [max_duration],
+	SUM(rs.avg_duration * rs.count_executions) / SUM(rs.count_executions) as avg_duration,
+	MAX(rs.max_num_physical_io_reads) as max_num_physical_io_reads,
+	SUM(rs.avg_num_physical_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_num_physical_io_reads,
+	MAX(rs.max_page_server_io_reads) as max_page_server_io_reads,
+	SUM(rs.avg_page_server_io_reads * rs.count_executions) / SUM(rs.count_executions) as avg_page_server_io_reads,
+	MAX(rs.max_log_bytes_used) as [max_log_bytes_used],
+	SUM(rs.avg_log_bytes_used * rs.count_executions) / SUM(rs.count_executions) as avg_log_bytes_used
+FROM sys.query_store_query q WITH (NOLOCK)
+	JOIN sys.query_store_plan p WITH (NOLOCK) ON q.query_id = p.query_id
+	JOIN sys.query_store_runtime_stats rs WITH (NOLOCK) ON p.plan_id = rs.plan_id
+	JOIN sys.query_store_runtime_stats_interval rsi WITH (NOLOCK) ON rs.runtime_stats_interval_id = rsi.runtime_stats_interval_id
+WHERE rsi.start_time >= @currIntervalStartDate AND rsi.end_time <= @currIntervalEndDate
+GROUP BY rs.plan_id, rs.execution_type, rs.runtime_stats_interval_id;
```

Comment:
I would add OPTION (RECOMPILE) to this query and the waits query below. As Query Store data grows and is purged, cached plans may become suboptimal. With the frequency of execution we have for these, the overhead is minimal.

Speaking of query execution time, do we have timeout set on the client? Sometimes, queries against QDS will end up with poor plans and take a very long time. We should protect against that with a query execution timeout.

Reply:
@dimitri-furman, I added option (recompile) to the queries. As for command timeout it hasn't been implemented in the plugin yet. And I don't think it's a good idea to impement command timeout support in this pull request. There is already issue #2824 and closed pull request #6677 related to adding timeout support.

Reply:
Agreed, that PR addresses this.